src/AutoRest.CSharp/Common/Output/Expressions/Snippets.New.cs

```diff
@@ -50,7 +50,7 @@ public static RequestContextExpression RequestContext(CancellationTokenExpressio
 
             public static ValueExpression RequestFailedException(BaseResponseExpression response) => Instance(Configuration.ApiTypes.RequestFailedExceptionType, response);
 
-            public static ResourceIdentifierExpression ResourceIdentifier(ValueExpression resourceData) => new(Instance(typeof(ResourceIdentifier), new MemberExpression(resourceData, "Id")));
+            public static ResourceIdentifierExpression ResourceIdentifier(ValueExpression id) => new(Instance(typeof(ResourceIdentifier), id));
```

Comment:
This method is only used in a scenario of xml serialization to serialize the ID in resource data that we are meant to create an expression of `new ResourceIdentifier(idString)`, the previous implemenation is wrong and misleading

Reply:
Here we could just make it simply since it is only calling the ctor of resource identifier